### ISSUE TITLE
[bug] bundle.resources tauri.config.json seems to do nothing on android.

### ISSUE THREAD

**Issue by kartonrad** [tauri-apps/tauri#10671]
### Describe the bug

The files specified:
```json
{
  "bundle": {
    "active": true,
    "targets": "all",
    "icon": [
      "icons/32x32.png",
      "icons/128x128.png",
      "icons/128x128@2x.png",
      "icons/icon.icns",
      "icons/icon.ico"
    ],
    "resources": {
      "resources/models/vosk-model-de-0.6/*": "resources/models/vosk-model-de-0.6",
      "resources/models/vosk-model-small-de-0.15/*": "resources/models/vosk-model-small-de-0.15"
    }
  }
}
```

Are nowhere to be found on the android filesystem.
I checked every Path on the PathResolver and output ever directory in a tree format.
What is going on? :(

### Full `tauri info` output

```text
[✔] Environment
    - OS: Windows 10.0.22631 X64
    ✔ WebView2: 127.0.2651.105
    ✔ MSVC:
        - Visual Studio Build Tools 2019
        - Visual Studio Community 2022
    ✔ rustc: 1.80.0 (051478957 2024-07-21)
    ✔ cargo: 1.80.0 (376290515 2024-07-16)
    ✔ rustup: 1.27.1 (54dd3d00f 2024-04-24)
    ✔ Rust toolchain: stable-x86_64-pc-windows-msvc (default)
    - node: 20.13.1
    - yarn: 1.22.22
    - npm: 10.5.2

[-] Packages
    - tauri [RUST]: 2.0.0-beta.22
    - tauri-build [RUST]: 2.0.0-beta.17
    - wry [RUST]: 0.40.1
    - tao [RUST]: 0.28.0
    - tauri-cli [RUST]: 2.0.0-beta.20
    - @tauri-apps/api [NPM]: 2.0.0-beta.13
    - @tauri-apps/cli [NPM]: 2.0.0-beta.20

[-] App
    - build-type: bundle
    - CSP: unset
    - frontendDist: ../build
    - devUrl: http://localhost:1420/
    - framework: Svelte
    - bundler: Vite
```

**Comment by kartonrad** [comment]
This is where the resources should end up according to the documentation.

![image](https://github.com/user-attachments/assets/243a31b3-0957-459c-afb4-7203bd2a6a21)

PathResolver::resource_dir() outputs ` /data/user/0/de.kartonrad./cache` for me. (i removed my app name so the url is valid in practice)

As you can see, `resources/models` is no where to be found under this directory:
```
AppCache
08-18 19:41:28.537 12840 12897 I RustStdoutStderr: Path: /data/user/0/de.kartonrad./cache
08-18 19:41:28.537 12840 12897 I RustStdoutStderr: └── WebView
08-18 19:41:28.537 12840 12897 I RustStdoutStderr:     ├── Crash Reports
08-18 19:41:28.537 12840 12897 I RustStdoutStderr:     ├── Crashpad
08-18 19:41:28.537 12840 12897 I RustStdoutStderr:     │   ├── attachments
08-18 19:41:28.537 12840 12897 I RustStdoutStderr:     │   ├── completed
08-18 19:41:28.537 12840 12897 I RustStdoutStderr:     │   ├── new
08-18 19:41:28.537 12840 12897 I RustStdoutStderr:     │   └── pending
08-18 19:41:28.538 12840 12897 I RustStdoutStderr:     ├── Default
08-18 19:41:28.538 12840 12897 I RustStdoutStderr:     │   └── HTTP Cache
08-18 19:41:28.538 12840 12897 I RustStdoutStderr:     │       ├── Cache_Data
08-18 19:41:28.538 12840 12897 I RustStdoutStderr:     │       │   ├── 0c40f8c12596af7b_0
08-18 19:41:28.613 12840 12898 E VoskAPI : Model():model.cc:122) Folder '/data/user/0/de.kartonrad./cache/resources/models/vosk-model-de-0.6' does not contain model files. Make sure you specified the model path properly in Model constructor. If you are not sure about relative path, use absolute path specification.
08-18 19:41:28.615 12840 12897 I RustStdoutStderr:     │       │   ├── 2e21fe3ff2aa2285_0
08-18 19:41:28.615 12840 12897 I RustStdoutStderr:     │       │   ├── 5b255b5e7ed90683_0
08-18 19:41:28.615 12840 12897 I RustStdoutStderr:     │       │   ├── cf148ac223ee9706_0
08-18 19:41:28.615 12840 12897 I RustStdoutStderr:     │       │   ├── fd65c59707383386_0
08-18 19:41:28.618 12840 12897 I RustStdoutStderr:     │       │   ├── index
08-18 19:41:28.619 12840 12897 I RustStdoutStderr:     │       │   └── index-dir
08-18 19:41:28.644 12840 12897 I RustStdoutStderr:     │       │       └── the-real-index
08-18 19:41:28.645 12840 12897 I RustStdoutStderr:     │       └── Code Cache
08-18 19:41:28.676 12840 12897 I RustStdoutStderr:     │           ├── js
08-18 19:41:28.677 12840 12897 I RustStdoutStderr:     │           │   ├── index
08-18 19:41:28.677 12840 12897 I RustStdoutStderr:     │           │   └── index-dir
08-18 19:41:29.001 12840 12897 I RustStdoutStderr:     │           │       └── the-real-index
08-18 19:41:29.012 12840 12897 I RustStdoutStderr:     │           └── wasm
08-18 19:41:29.020 12840 12897 I RustStdoutStderr:     │               ├── index
08-18 19:41:29.020 12840 12897 I RustStdoutStderr:     │               └── index-dir
08-18 19:41:29.030 12840 12897 I RustStdoutStderr:     │                   └── the-real-index
08-18 19:41:29.072 12840 12897 I RustStdoutStderr:     ├── SafeBrowsing
08-18 19:41:29.072 12840 12897 I RustStdoutStderr:     └── font_unique_name_table.pb
```

**Comment by kartonrad** [comment]
Using resources with android doesn't seem ideal, as all "resources" in the android context are usually permanently bundled into the apk

But i think embedding defined resources using /assets or /raw folders under /gen/android and then writing them to the /cache directory on installation is the way to go here.

It would be increadibly stupid to embed large files into the rust executable, that are 3 gb that do not need to be in ram

**Comment by kartonrad** [comment]
And in my case, i really do have to read from the file system, since the library i'm using just acceots a path, not in memory objects :(

**Comment by kartonrad** [comment]
Having worked around this in Kotlin, i think it makes sense why this isn't implemented, since androids resource/asset mechanism is in the Java World and it would probably require some effort to tap into it.
Just copying from APK to an Internal Storage Path would waste space on the users device.

That being said, it would be *great* if it was documented that tauris resource bundling does not work on mobile.

**Comment by FabianLars** [comment]
> That being said, it would be great if it was documented that tauris resource bundling does not work on mobile.

Agreedl, whether or not we actually (can) add the feature on mobile or not since it won't happen in the initial 2.0 release.

/upstream tauri-apps/tauri-docs

**Comment by lucasfernog** [comment]
closed by #10696
